react/features/notifications/components/NotificationsContainer.web.js

```diff
@@ -0,0 +1,159 @@
+import { FlagGroup } from '@atlaskit/flag';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { hideNotification } from '../actions';
+
+/**
+ * The duration for which a notification should be displayed before being
+ * dismissed automatically.
+ *
+ * @type {number}
+ */
+const DEFAULT_NOTIFICATION_TIMEOUT = 2500;
+
+/**
+ * Implements a React {@link Component} which displays notifications and handles
+ * automatic dismissmal after a notification is shown for a defined timeout
+ * period.
+ *
+ * @extends {Component}
+ */
+class NotificationsContainer extends Component {
+    /**
+     * {@code NotificationsContainer} component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The notifications to be displayed, with the first index being the
+         * notification at the top and the rest shown below it in order.
+         */
+        _notifications: React.PropTypes.array,
+
+        /**
+         * Invoked to update the redux store in order to remove notifications.
+         */
+        dispatch: React.PropTypes.func
+    };
+
+    /**
+     * Initializes a new {@code NotificationsContainer} instance.
+     *
+     * @param {Object} props - The read-only React Component props with which
+     * the new instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        /**
+         * The timeout set for automatically dismissing a displayed
+         * notification. This value is set on the instance and not state to
+         * avoid additional re-renders.
+         *
+         * @type {number|null}
+         */
+        this._notificationDismissTimeout = null;
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onDismissed = this._onDismissed.bind(this);
+    }
+
+    /**
+     * Sets a timeout if the currently displayed notification has changed.
+     *
+     * @inheritdoc
+     * returns {void}
+     */
+    componentDidUpdate() {
+        const { _notifications } = this.props;
+
+        if (_notifications.length && !this._notificationDismissTimeout) {
+            const notification = _notifications[0];
+            const { options, uid } = notification;
+            const timeoutDuration
+                = (options && options.timeout) || DEFAULT_NOTIFICATION_TIMEOUT;
```

Comment:
The only thing we're currently handling as option is options.timeout and I've seen this used only once, so can we make this a prop and don't handle options at all for now, until we need something more generic?